query/src/main/scala/filodb/query/util/LogicalPlanUtil.scala

```diff
@@ -0,0 +1,56 @@
+package filodb.query.util
+
+import filodb.core.query
+import filodb.core.query.{ColumnFilter, Filter}
+import filodb.query._
+
+object LogicalPlanUtil {
+
+  private def getFilterValue(filter: Filter): Set[String] = {
+    filter match {
+      case f: query.Filter.Equals => Set(f.value.toString)
+      case f: query.Filter.And => getFilterValue(f.left).union(getFilterValue(f.right))
+      case f: query.Filter.In => f.values.map(_.toString)
+      case f: query.Filter.NotEquals => Set(f.value.toString)
+      case f: query.Filter.NotEqualsRegex => Set(f.value.toString)
+      case f: query.Filter.EqualsRegex => Set(f.value.toString)
+    }
+  }
+
+ private def getLabelFromFilters(filters: Seq[ColumnFilter], metricColumnName: String): Option[Set[String]] = {
+   val matchingFilters = filters.filter(_.column.equals(metricColumnName))
+    if (matchingFilters.isEmpty)
+      None
+    else
+      Some(getFilterValue(matchingFilters.head.filter))
+  }
+
+  def getLabelNameFromLogicalPlan(logicalPlan: LogicalPlan, labelValue: String): Option[Set[String]] = {
```

Comment:
Method name is not correct, since it returns the values from filter and not the labelName

Reply:
Good catch! I have made the change